File: 프로그래머스/lv1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    const filteredReserve =  reserve.filter((v) => !lost.includes(v)).sort((a, b) => a - b);
    const filteredLost = lost.filter((v) => !reserve.includes(v)).sort((a, b) => a - b);
    
    return n - filteredLost.map((v) => {
        const borrowFromFront = filteredReserve.indexOf(v - 1);
        const borrowFromBack = filteredReserve.indexOf(v + 1);
        
        if (borrowFromFront !== -1) {
            filteredReserve.splice(borrowFromFront, 1);
            return null;
        } else if (borrowFromBack !== -1) {
            filteredReserve.splice(borrowFromBack, 1);
            return null;
        } else {
            return v;    
        }
        
    }).filter((v) => v).length;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    filtered_reserve = sorted([i for i in reserve if i not in lost])
    filtered_lost = sorted([i for i in lost if i not in reserve])
    
    answer = n - len(filtered_lost) 
    
    for l in filtered_lost:
        if l - 1 in filtered_reserve :
            filtered_reserve.remove(l - 1)
            answer += 1
            continue
            
        if l + 1 in filtered_reserve:
            filtered_reserve.remove(l + 1)
            answer += 1
            continue
    
    return answer